.husky/pre-commit

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+. "$(dirname "$0")/_/husky.sh"
+
+echo 'Formatting Code'
+
+# Check if CSharpier is installed
+if ! dotnet tool list --local | grep -q "csharpier"; then
+    echo "Installing csharpier..."
+    dotnet tool install --local csharpier
+fi
+
+# Get only modified C# files
+CHANGED_FILES=$(git diff --name-only --diff-filter=d HEAD | grep '\.cs$' || true)
+
+# If there are modified files, run CSharpier, otherwise skip formatting.
+if [ -n "$CHANGED_FILES" ]; then
+    echo "Running CSharpier on changed files..."
+    dotnet csharpier $CHANGED_FILES || {
+        echo "CSharpier detected unformatted files. Please run 'dotnet csharpier .' to fix."
+        exit 1
+    }
+else
+    echo "No C# files changed, skipping formatting."
+fi
```

Comment:
I was surprised that this bash script worked while running in Powershell. But it did work perfectly 😄 . I am watching this because we probably should apply this tool in projects on the Tools team as well.